### ISSUE TITLE
spellicon & rank only shown when cast from actionbar.

### ISSUE THREAD

**Issue by Larsy93** [Attero/Punschrulle#2]
Does not show if spell is cast directly from spellbook, or more importantly from a macro.

**Comment by Attero** [comment]
This has to do with limitations in the client. To my knowledge there are no clean ways of verifying what spell was actually cast and when in either of those cases.

There are a couple of imperfect workarounds that can be implemented, but all of them bring with them other issues. It's possible you will see a workaround allowing most macros to show rank and icon in the next few days. The spellbook is very low priority however.

**Comment by Larsy93** [comment]
Yeah the spellbook idc but macros would be sweet since im a healer and seing spellranks from my mouseover macros would be very beneficial

**Comment by Attero** [comment]
Added spellrank + icons to CASTS, but not channels, from macros as well as spellbook. If you are still having any issues with macros, please attach the macro in question.

Channels are a little bit trickier, but if there is demand i may look into a workaround for that as well at a later point.

**Comment by Larsy93** [comment]
Hi, when I have the new version enabled the macro does not cast at all when I try to use it, when i disable punschrulle it works fine, so something must have broken it in the last update.

Here is the macro

/script if nil then CastSpellByName("Healing Wave") end
/lunamo Healing Wave

and /lunamo is the mouseover macro thingy for Luna Unitframes.

**Comment by Attero** [comment]
I fixed a related bug that messed with the spelltargeting of luna, but never had issues with spells refusing to cast.

If you still have issues, could you try it with only Luna and Punschrulle enabled and see if the problem persists? Otherwise it's probably an interaction with some third addon, and if so I need to know which.

**Comment by Larsy93** [comment]
Home again after new years, will report back with info, will test now.

**Comment by Larsy93** [comment]
It happens with only luna & punschrulle enabled

this is the error i get
https://i.gyazo.com/ecde99248b27ce19e0c6799c9a605740.png

this is the macro

/script if nil then CastSpellByName("Healing Wave") end
/lunamo Healing Wave

**Comment by Larsy93** [comment]
The character just stands still, no reaction whatsoever when pressing the macro.

**Comment by Attero** [comment]
Just pushed a change that should have fixed it.

**Comment by Larsy93** [comment]
That was fast, works like charm now, thank you so much. Late Happy New Year 👍
